### PR TITLE
implement aliasByMetric

### PR DESCRIPTION
**Pull request by vaguiar**
**Describe your changes**
Native implementation of aliasByMetric

**Testing performed**
Unit tests and integration testing

### CODE REVIEW

File: expr/func_aliasbymetric.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+func (s *FuncAliasByMetric) Exec(dataMap DataMap) ([]models.Series, error) {
+	series, err := s.in.Exec(dataMap)
+	if err != nil {
+		return nil, err
+	}
+	for i, serie := range series {
+		n := extractMetric(serie.Target)
+		series[i].Target = n
+		series[i].QueryPatt = n
```
Comment (vaguiar) [inline expr/func_aliasbymetric.go:33]: Note that unlike https://github.com/grafana/metrictank/blob/master/expr/func_aliasbynode.go#L32-L36, I'm not resetting the tags on the output series

File: expr/func_aliasbymetric_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		},
+		t,
+	)
+}
+
+func TestAliasByMetricWithTags(t *testing.T) {
+	/* Long metric string with multiple tag values
+	   which can accept chars like [a-zA-Z0-9-_./%@ +<>!]
+	*/
+	var longMetric string = "report.runtime.d.max;cluster=tkap*;datacenter=!or@ngeb%u<>rg1;gutsReportName=guts-timeseries;gutsReporterJobType=localJob;host=tel-kap-pob-r03-n02.bloomberg.com;namespace=gutsReporting;operatingSystem=redhat;operatingSystemVer=7.6;parentCluster=tlmtry;stage=s4;subStage=s4d;timezone=UTC"
```
Comment (vaguiar) [inline expr/func_aliasbymetric_test.go]: I will mask the tag values here or use generic ones

**Comment by agao48** [comment]
don't forget to update https://github.com/grafana/metrictank/blob/master/docs/graphite.md

**Review by agao48 — APPROVED** [review]

**Review by shanson7 — CHANGES_REQUESTED** [review]
Also update docs/graphite.md

**Review by shanson7 — APPROVED** [review]

**Pull request merged**